### ISSUE TITLE
QueryFunc is leaking errors from grep.

### ISSUE THREAD

**Issue by dschaper** [pi-hole/pi-hole#1805]
**In raising this issue, I confirm the following:** `{please fill the checkboxes, e.g: [X]}`

- [x] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md).
- [x] The issue I am reporting can be *replicated*.
- [x] The issue I am reporting isn't a duplicate (see [FAQs](https://github.com/pi-hole/pi-hole/wiki/FAQs), [closed issues](https://github.com/pi-hole/pi-hole/issues?utf8=%E2%9C%93&q=is%3Aissue%20is%3Aclosed%20), and [open issues](https://github.com/pi-hole/pi-hole/issues)).

**How familiar are you with the the source code relevant to this issue?:**

`8`

---
**Expected behaviour:**

`pihole -q` should just return usable information

**Actual behaviour:**

```
root@ubuntu-512mb-nyc3-01:/var/log# pihole -q -all google.com
grep: blacklist.txt: No such file or directory
grep: *.domains: No such file or directory
  [i] No results found for google.com found within block lists
```

**Steps to reproduce:**

`pihole -q`

**Comment by Mcat12** [comment]
Introduced by https://github.com/pi-hole/pi-hole/commit/c2d3e99ddd7a3fbe65d2cfc2ebb8e5ad917b78de

**Comment by DL6ER** [comment]
While we're at it, the whole `query` functionality should be moved into a separate file as we did also with nearly all other functions that are bulkier than a few lines.

**Comment by dschaper** [comment]
Fixed in #2234

**Comment by jacobsalmela** [comment]
This is fixed in v4.0
